### PR TITLE
Add a parse function to parse a comma separated list

### LINKED ISSUE #77: [Feature Request] Support comma separated range list in RangeParser
Currently you can parse one range string like

```rust
assert_eq!(RangeParser::parse_one("KK+").unwrap().len(), 12)
```

I'd like to have the ability to parse a broader range string, something like `JJ+, A2s+`:

```rust
assert_eq!(RangeParser::parse_one("KK+,A2s+").unwrap().len(), 60)
```

Or a bit more complicated

```rust
assert_eq!(RangeParser::parse_one("AK-87s,A2s+").unwrap().len(), 72)
```

Since `AK-87s` overlaps `AKs` with `A2s+` it would need to look out for duplicates.

Do you feel this could be a useful addition?

### PR DESCRIPTION
**Pull request by JohannesKlauss**
This fixes #77 
Not sure if having both functionalities in one function would be better.

Let me know if I should change anything 🙂 
Thanks for the great library!

### CODE REVIEW

File: src/holdem/parse.rs [hunk 1]
```diff
@@ -522,6 +523,33 @@ impl RangeParser {
...
+    /// ```
+    /// use rs_poker::holdem::RangeParser;
+    ///
+    /// let hand = RangeParser::parse("KK+,A2s+").unwrap();
+    /// assert_eq!(60, hand.len());
+    ///
+    /// // Filters out duplicates.
+    /// assert_eq!(RangeParser::parse_one("AK-87s,A2s+").unwrap().len(), 72)
+    /// ```
+    pub fn parse(r_str: &str) -> Result<Vec<Hand>, RSPokerError> {
```
Comment (elliottneilclark) [inline src/holdem/parse.rs:542]: What about `parse_all` or `parse_any`? The other naming is explicit about the number of hands to expect so we should continue that pattern.

File: src/holdem/parse.rs [hunk 2]
```diff
@@ -522,6 +523,33 @@ impl RangeParser {
...
+        let mut set: HashSet<Hand> = HashSet::new();
+
+        for hand_str in r_str.split(',') {
+            let hand = RangeParser::parse_one(hand_str.trim())?;
+
+            set.extend(hand)
+        }
```
Comment (elliottneilclark) [inline src/holdem/parse.rs:549]: I think this is better written as split -> map(parse_one) -> collect into HashSet -> into_iter -> collect as vec()

See https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect For error handling.

**Review by elliottneilclark — CHANGES_REQUESTED** [review]

File: src/holdem/parse.rs (hunk 1 above)
Comment (JohannesKlauss) [reply src/holdem/parse.rs:542]: Maybe `parse_many`?

Comment (elliottneilclark) [reply src/holdem/parse.rs:542]: Sounds good to me.

File: src/holdem/parse.rs (hunk 2 above)
Comment (JohannesKlauss) [reply src/holdem/parse.rs]: I changed the implementation to try and follow your suggestions. I am still relatively new to rust, so let me know if this needs improvement.

File: src/holdem/parse.rs [hunk 3]
```diff
@@ -522,6 +523,42 @@ impl RangeParser {
...
+
+        match errors.is_empty() {
+            true => {
+                set.extend(hands);
+
+                Ok(set.into_iter().collect())
+            }
+            false => Err(errors),
+        }
+    }
```
Comment (elliottneilclark) [inline src/holdem/parse.rs:561]: We should only return RSPokerError. What do you think about this?

```suggestion
    pub fn parse_many(r_str: &str) -> Result<Vec<Hand>, RSPokerError> {
        let all_hands: Vec<_> = r_str
            // Split into different ranges
            .split(',')
            // Try to parse the ranges.
            .map(|s| RangeParser::parse_one(s.trim()))
            // Use FromIterator to get a result and unwrap it.
            .collect::<Result<Vec<_>, _>>()?;

        // Filter the unique hands.
        let unique_hands: HashSet<Hand> = all_hands.into_iter().flatten().collect();

        // Transform hands into a vec for storage
        Ok(unique_hands.into_iter().collect())
    }
```

**Review by elliottneilclark — APPROVED** [review]

**Comment by elliottneilclark** [comment]
Thank you for the added usability @JohannesKlauss

**Pull request merged**